src/app/components/client/ExposuresFilter.tsx

```diff
@@ -186,6 +191,14 @@ export const ExposuresFilter = ({
     </form>
   );
 
+  // Reset current filter state if selection has not been confirmed.
+  useEffect(() => {
+    const shouldResetFilterState = filterValues !== filterState;
+    if (!filterDialogState.isOpen && shouldResetFilterState) {
+      setFilterState(filterValues);
+    }
+  }, [filterDialogState, filterValues, filterState]);
+
```

Comment:
This feels wrong to me, since no [side effects](https://en.wikipedia.org/wiki/Side_effect_(computer_science)) happen here. Wouldn't it also work to pass an `onOpenChange` callback to `useOverlayTriggerState`, which calls `setFilterState(filterValues)` whenever the popover closes?

Reply:
Thanks for the suggestion — moved to `onOpenChange` in [0689e3b](https://github.com/mozilla/blurts-server/pull/3310/commits/0689e3b9f0d2b6add695327971a197df96b0bfc6).